README.md

```diff
@@ -155,29 +158,29 @@ gets to be confirmed `getBalance` will return 40KIN and `getPendingBalance` will
 Similarly if you have 30KIN and someone else transfer 2KIN to you, until the transaction gets to be confirmed
 `getBalance` will return 30KIN and `getPendingBalance` will return 32KIN.
 ```java
-account.getPendingBalance(new ResultCallback<Balance>() {
-    
-    @Override
-    public void onResult(Balance result) {
-        Log.d("example", "The balance is: " + result.toString());
-    }
+Request<Balance> balanceRequest = account.getPendingBalance();
+balanceRequest.run(new ResultCallback<Balance>() {
 
     @Override
-    public void onError(Exception e) {
-        e.printStackTrace();
-    }
+        public void onResult(Balance result) {
+            Log.d("example", "The balance is: " + result.toString());
+        }
+
+        @Override
+        public void onError(Exception e) {
+            e.printStackTrace();
+        }
 });
 ```
 
 ### Sync vs Async
-As you have seen above we have provided a very simple solution for accessing the ethereum network on a background thread.
-In this solution `getBalance`, `getPendingBalance` and `sendTransaction` methods, run on a background thread 
-and accesses the Ethereum network. The `onResult` and `onError` callback methods are executed on the android UI thread.
-This solution suffers from inability to cancel the tasks and you need to use it very carefully to avoid leaking Activity context.
-An example of how this can be used without leaking context can be found in our [Sample App](sample/).
-
-We will be providing with a better solution shortly. In the meantime you are welcome to use the synchronous version of `getBalance`, 
-`sendTransaction` and `getPendingBalance` methods making sure you call them in a background thread in any way that you are accustomed.
+
+Asynchronous requests supported by `Request` object, that run on serial background thread, 
```

Comment:
Asynchronous requests are supported by our `Request` object. The `request.run()` method will perform the request on a serial background thread and notify success/failure using `ResultCallback` on the android main thread.  In addition, `cancel(boolean)` method can be used to safely cancel requests and detach callbacks.

Reply:
updated